Editor/Window/Components/ProfileSelector.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Linq;
+using Editor.CoreAPI;
+using Editor.Resources.EditorWindow;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    public class ProfileSelector : VisualElement
+    {
+        public new class UxmlFactory : UxmlFactory<ProfileSelector> { }
+
+        private readonly StateManager _stateManager;
+        private readonly TextProvider _textProvider;
+
+        private DropdownField _dropdown => this.Q<DropdownField>("Dropdown");
+        private Label _bucketName => this.Q<Label>("BucketName");
+        private Label _region => this.Q<Label>("Region");
+        private Label _status => this.Q<Label>("BootstrapStatus");
+
+        public ProfileSelector()
+        {
+            var asset = Resources.Load<VisualTreeAsset>("EditorWindow/Components/ProfileSelector");
+            asset.CloneTree(this);
+
+            LocalizeText();
+
+            _stateManager = EditorWindow.GetWindow<GameLiftPlugin>().StateManager;
+            _stateManager.OnProfileSelected += UpdateGUI;
```

Comment:
Does this cause trouble when changing between tabs (because it will theoretically be on each tab)?
* Does it add extra callbacks with each tab change?
* Is there risk for hidden components still having these callbacks occur?

Is there an OnEnable/OnDisable option for VisualElements that could solve this problem for us?

Reply:
The elements will all be on the same page technically and just hidden/shown through css.
- Hidden elements will still update
- Changing tab will not cause anything to update
- There is not OnEnable/Disable for VisualElements, only EditorWindow

Reply:
Ack